### PR TITLE
Java 7 changes

### PR DESCRIPTION
**Pull request by fge**
This has required changing the parent pom version to 19.

But this way the calls to Closeables.closeQuietly() could be removed.

Also, while we are at it, replace all invocations of:
- Preconditions.checkNonNull() with Objects.requireNonNull();
- Lists.newArrayList(...) with new ArrayList<>() (where appropriate).

Also update Collections usage.

### CODE REVIEW

**Comment by Godin** [comment]
Thanks for PR, it will be reviewed during next sprint on SSLR.

**Comment by Godin** [comment]
This PR is going to be part of upcoming sprint, associated ticket - http://jira.sonarsource.com/browse/SSLR-370

**Comment by Godin** [comment]
The rest, plus replacements for "Sets.newHashSet" and "Maps.newHashMap", except "Closeables" and "SafeVarargs", done in c395cf70a84cbdb806ee9004025611a76726334a

**Comment by Tzrlk** [comment]
Is this going to be merged and released at some point?

**Comment by Godin** [comment]
@Tzrlk could you please elaborate:
*  which exact part of this PR has interest for you, since some parts of it were already released wit version 1.21?
* and why they are important for you, since most of this PR - are non-functional cosmetic changes?

**Comment by Tzrlk** [comment]
The key point is removing the use of `closeables.closeQuietly`, so the version of guava included in the Java plugin and API doesn't conflict with sslr's dependency on v10

**Comment by Godin** [comment]
@Tzrlk honestly, I'm not aware of problems in Java Plugin caused by this.
@benzonico Nico, could you please confirm/deny?

**Comment by Tzrlk** [comment]
The issue isn't with the Java plugin as such, but custom plugins built of it.

**Comment by benzonico** [comment]
@Tzrlk I saw your stackoverflow question : http://stackoverflow.com/questions/39239840/how-do-i-fix-sonarqube-guava-version-incompatibility-closequietly-in-my-projec 

Could you please share a minimalistic example with the version of each component used to reproduce the issue so we can actually try to understand what is going on ? 
Right now you have a handful of claims but with nothing really tangible to back up those so we can actually try to investigate and fix the issue.

We are struggling to understand why the closeQuietly references in SSLR could have an impact on custom rule and given the stacktrace on SO, it has nothing to do with it (but this is a wild guess as we have very little information to reason on). 

TL;DR: Please post a minimal reproducer so we can investigate the issue. Reach us out on sonarqube@googlegroups.com

Thanks
